scripts/Microsoft.DotNet.Cli.Build.Framework/TargetConditions/EnvironmentAttribute.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Cli.Build.Framework
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
+    public class EnvironmentAttribute : TargetConditionAttribute
+    {
+        private string _envVar;
+        private string _expectedVal;
+
+        public EnvironmentAttribute(string envVar, string expectedVal)
+        {
+            if (string.IsNullOrEmpty(envVar))
+            {
+                throw new ArgumentNullException("envVar");
+            }
+
+            _envVar = envVar;
+            _expectedVal = expectedVal;
+        }
+
+        public override bool EvaluateCondition()
+        {
+            var actualVal = Environment.GetEnvironmentVariable(_envVar);
+
+            if (string.IsNullOrEmpty(_expectedVal))
+            {
+                return string.IsNullOrEmpty(actualVal) ||
```

Comment:
This is saying `string.Empty`, `null`, `0`, and `false` are all equivalent as environment variable values?

I would prefer for values to be matched explicitly, what do we gain from this?